eras/conway/impl/src/Cardano/Ledger/Conway/PParams.hs

```diff
@@ -113,7 +113,6 @@ import Cardano.Ledger.HKD (
 import Cardano.Ledger.Plutus.CostModels (
   CostModel,
   decodeCostModelFailHard,
```

Comment:
I see that `decodeCostModelFailHard` is still used in this module as well as `Cardano.Ledger.Plutus.Evaluate`. Why do we still need this in Conway?

Reply:
I think the name is a bit too harsh sounding, because it will work fine with variable number of parameters. This is a decoder that will fail for other reasons and it only works for known plutus language versions. We do need it in the ConwayGenesis (usage in `Cardano.Ledger.Conway.PParams` module) and during debugging plutus scripts (usage in `Cardano.Ledger.Plutus.Evaluate`) because at those.

I think I should just rename it to something less scary sounding, something like: `decodeCostModel` :smile:

Reply:
`decodeCostModelFailHard` is now deprecated in favor of `decodeCostModel`